To-do-list/Controller/ViewController.swift

```diff
@@ -9,94 +9,56 @@ import UIKit
 
 class ViewController: UIViewController {
     
-    @IBOutlet weak var tableView: UITableView!
-    
+    private var mainViewModel: MainViewModel?
     private let taskManager = TaskManager()
     
+    @IBOutlet weak var tableView: UITableView!
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        tableView.dataSource = self
-        tableView.delegate = self
+        tableView?.dataSource = self
+        setNavBar()
     }
     
-    func refresh() {
-        let taskDictionnary = UserDefaults.standard.object(forKey: "newTask") as? [String: String] ?? [String: String]()
-        guard let titleString = taskDictionnary["title"] else { return }
-        guard let descriptionString = taskDictionnary["description"] else { return }
-        taskManager.addTask(title: titleString, description: descriptionString)
-        
-        taskManager.reloadTableView(tableView: tableView)
+    func reloadTableView() {
+        self.tableView?.reloadData()
     }
     
-    func openTaskEdition() {
-        guard let taskViewController = storyboard?.instantiateViewController(withIdentifier: "TaskViewController") as? TaskViewController else { return }
-        
-        taskViewController.completionHandler = {
-            self.refresh()
-        }
-        navigationController?.pushViewController(taskViewController.self, animated: true)
+    func configure(mainViewModel: MainViewModel) {
+        self.mainViewModel = mainViewModel
     }
     
-    func deleteAll() {
-        taskManager.tasks.removeAll()
-        taskManager.reloadTableView(tableView: tableView)
+    @objc func addTask() {
+        mainViewModel?.shouldOpenTaskViewController()
```

Comment:
let's use function viewModel.addTask()

Reply:
Ok.., didn't understand here

Reply:
whta is shouldOpenTaskViewController?

Reply:
It's when pressing the plus button the main viewController opens the taskViewController via Appdelegate